security/generate.go

```diff
@@ -104,42 +106,49 @@ func Generate(genpkg string, roots []eval.Root, files []*codegen.File) ([]*codeg
 // context gets initialized with the security requirements.
 func Example(genpkg string, roots []eval.Root, files []*codegen.File) ([]*codegen.File, error) {
 	var (
-		data   *ServiceData
+		data   []*ServiceData
 		apiPkg string
 	)
 	for _, root := range roots {
 		switch r := root.(type) {
 		case *goadesign.RootExpr:
 			apiPkg = strings.ToLower(codegen.Goify(r.API.Name, false))
 			for _, s := range r.Services {
-				data = BuildSecureServiceData(s)
+				sd := BuildSecureServiceData(s)
```

Comment:
Should we make `BuildSecureServiceData` accept an extra argument so we can pass in `apiPkg` and not have to set it below?